### ISSUE TITLE
Apitrace does not create traces in recent version of wine+Nine 

### ISSUE THREAD

**Issue by EoD** [apitrace/apitrace#398]
This is a follow-up of #294

The recent version of [wine+nine](https://github.com/iXit/wine) use a separate dll "d3d9-nine.dll" to make nine fully optional and get it ready to be merged into wine-staging. This seems to cause a problem with apitrace's dll injection and hence does not generate any more traces:
```
info: did not find D:\Downloads\apitrace-msvc\x86\bin\..\wrappers\RelWithDebInfo\d3d9.dll
info: found D:\Downloads\apitrace-msvc\x86\bin\..\lib\wrappers\d3d9.dll
inject -D D:\Downloads\apitrace-msvc\x86\bin\..\lib\wrappers\d3d9.dll -- launcher.exe
fixme:module:load_dll Loader redirect from L"D:\\Downloads\\apitrace-msvc\\x86\\bin\\..\\lib\\wrappers\\d3d9.dll" to L"d3d9-nine.dll"
fixme:iphlpapi:NotifyAddrChange (Handle 0x33cbac, overlapped 0x2f9bbe0): stub
```

Could this be circumvented by apitrace or does this need to be addressed in wine?

**Comment by jrfonseca** [comment]
> Could this be circumvented by apitrace or does this need to be addressed in wine?

WINE is not respecting the `WINEDLLOVERRIDES` when it forcibly replaces the wrapper with d3d9-nine.dll:

    fixme:module:load_dll Loader redirect from L"D:\\Downloads\\apitrace-msvc\\x86\\bin\\..\\lib\\wrappers\\d3d9.dll" to L"d3d9-nine.dll"

This could be circumvented by apitrace, by renaming the D3D9 wrapper something other than `d3d9.dll`, e.g., `d3d9trace.dll`.  However, being able to manually copy `d3d9.dll` wrapper into the app directory is very useful sometimes (for apps which DLL injection doesn't work well), so I'd like to keep the wrapper DLL name if possible.

**Comment by EoD** [comment]
If you don't want to rename the wrapper dll, could you provide an argument which allows specifying a custom dll name? If I understood you correctly, this would fix the issue.

**Comment by jrfonseca** [comment]
> If you don't want to rename the wrapper dll, could you provide an argument which allows specifying a custom wrapper-dll name?

Sorry, but this is not practical.  Note that when injecting the DLL, the wrapper is not copied, or renamed, or anything. It just stays there wherever it is.

So what you're asking is a lot of special code just for Wine.

Wine is not something that I'm directly interested in. I already spent time to figure out a way for apitrace to work with Wine and wrote up the instructions on https://github.com/apitrace/apitrace/wiki/WINE . I'm afraid the ball is now on Wine community's court to keep it working and not breaking it, especially given the Wine community are those benefiting the most.  I have a never ending pile of 3rd party patches and unfinished branches for apitrace to take care of.

In short, I'm afraid you'll just need to figure this one out by yourselves.
